src/nspawn/nspawn.c

```diff
@@ -2317,7 +2317,11 @@ static int drop_capabilities(uid_t uid) {
 
                 if (q.ambient == (uint64_t) -1 && ambient_capabilities_supported())
                         q.ambient = 0;
-        } else
+
+                if (capability_quintet_mangle(&q))
+                        return log_error_errno(EPERM, "Cannot set capabilities that are not in the bounding set.");
```

Comment:
needs SYNTHETIC_ERRNO() around the EPERM please.

(that macro indicates that we generate the error here, instead of receiving it)

Reply:
"bounding set" → "current bounding set" maybe?